cmd/kubeops/internal/handler/dashboardcompat.go

```diff
@@ -17,9 +19,17 @@ import (
 )
 
 func newDashboardCompatibleRelease(h3r h3.Release) h2.Release {
+	var deleted *timestamp.Timestamp
+	if !h3r.Info.Deleted.IsZero() {
+		var err error
+		deleted, err = ptypes.TimestampProto(h3r.Info.Deleted.Time)
+		if err != nil {
+			log.Errorf("Failed to parse deletion time %v", err)
```

Comment:
This should, imo, either be a failure (preferred) or we set a deleted time (some epoch) so that the UI at least knows it has been deleted. I don't think ignoring errors (because it's easier than pushing errors up the stack) is going to save us time longer-term.

On another line of thought: why is the UI depending on the timestamp, given that if we have a record of a deleted release, then (I think?) by definition, we have a record which can be purged? (as once it is purged, we have no record of the deleted release).

+1, but see what you think.

Reply:
> This should, imo, either be a failure (preferred) or we set a deleted time (some epoch) so that the UI at least knows it has been deleted. I don't think ignoring errors (because it's easier than pushing errors up the stack) is going to save us time longer-term.

I didn't want to break the structure of the code (and it's unlikely that this function would fail) but okay, we can return the error.

> On another line of thought: why is the UI depending on the timestamp, given that if we have a record of a deleted release, then (I think?) by definition, we have a record which can be purged? (as once it is purged, we have no record of the deleted release).

The UI depends on the `status.deleted` field to be populated (with any info) to identify a release as deleted. That's the only identifier that we have (or at least had in helm 2, I am not sure if something new is available) to know if a release is deleted.

Reply:
Ah - I see. I'd misread and thought the code already knew this was deleted, but see it's based on the timestamp itself.